### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by senilio** [hoihrig/prometheus-tellstick-exporter#1]
Hi!

I have a Tellstick Znet Lite V2. I tried your exporter today since I'm migrating from influxdb to prometheus, but unfortunately I ran into issues. Host server is an Ubuntu 16 server.

Exporter starts just fine, and I can load http://192.168.44.20:9317, but once I click the Metrics link on the page, the exporter crashes with the below output.

```
root@monitor:~/telldus-exporter# /usr/local/bin/tellstick_exporter -local-tellstick-host "192.168.44.20" -telldus-token "eyJhbGciOiJIU...<snip>...za1bwlhP6QZGWs2A5Q"
INFO[0000] Metric Descriptions added!

ERRO[0080] Get 192.168.44.20/api/devices/list: unsupported protocol scheme ""
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x6b4e05]

goroutine 36 [running]:
github.com/hoihrig/prometheus-tellstick-exporter/telldus_local.makeRequest(0x7fffffeff7e8, 0xd, 0x7fffffeff805, 0xab, 0x768ea6, 0xc, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/hoihrig/prometheus-tellstick-exporter/telldus_local/telldus_local_connector.go:85 +0x305
github.com/hoihrig/prometheus-tellstick-exporter/telldus_local.getDeviceCount(0x7fffffeff7e8, 0xd, 0x7fffffeff805, 0xab, 0x3)
        /go/src/github.com/hoihrig/prometheus-tellstick-exporter/telldus_local/telldus_local_connector.go:60 +0x7c
github.com/hoihrig/prometheus-tellstick-exporter/telldus_local.(*Exporter).Collect(0xc4200de000, 0xc42005cba0)
        /go/src/github.com/hoihrig/prometheus-tellstick-exporter/telldus_local/prometheus.go:21 +0x5f
github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func2(0xc420013ed0, 0xc42005cba0, 0x8dd940, 0xc4200de000)
        /go/src/github.com/prometheus/client_golang/prometheus/registry.go:383 +0x61
created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather
        /go/src/github.com/prometheus/client_golang/prometheus/registry.go:384 +0x2ec
```

Any help is appreciated! Thanks.

**Comment by hoihrig** [comment]
Hi,

thanks for reporting this. I think it is because your --local-tellstick-host parameter lacks the protocol. Please try with --local-tellstick-host=http://192.168.44.20

It should not have crashed this hard though

**Comment by senilio** [comment]
This did the trick... now that I'm reading the help text again I notice it even says "Host to connect to *including protocol.*"

Thanks for this 👍

**Comment by senilio** [comment]
Straying a bit from the original issue... I am getting output from the exporter now, but any value is 0. The sensors are properly detected though:

```
......
# HELP telldus_sensors_count Number of Sensors
# TYPE telldus_sensors_count gauge
telldus_sensors_count 6
# HELP telldus_sensors_metric Value of the Sensor
# TYPE telldus_sensors_metric gauge
telldus_sensors_metric{Sensorname="Power",type="watt",unit="KWh"} 0
telldus_sensors_metric{Sensorname="Outside",type="temp",unit="Celsius"} 0
telldus_sensors_metric{Sensorname="Kitchen",type="temp",unit="Celsius"} 0
telldus_sensors_metric{Sensorname="Basement",type="temp",unit="Celsius"} 0
telldus_sensors_metric{Sensorname="Office",type="watt",unit="Watt"} 0
telldus_sensors_metric{Sensorname="Ute",type="temp",unit="Celsius"} 0
```

This is the config block I'm using in prometheus.yml:

```
  - job_name: 'tellstick_exporter'
    metrics_path: /metrics
    static_configs:
      - targets: ['localhost:9317']
```

This is an example of sensor output when I query the API using curl:

```
{
  "data": [
    {
      "name": "temp",
      "scale": 0,
      "value": 18.899999999999999
    },
    {
      "name": "humidity",
      "scale": 0,
      "value": 42.0
    }
  ],
  "id": 10,
  "model": "temperaturehumidity",
  "name": "Outside",
  "protocol": "fineoffset",
  "sensorId": 10
}
```

What am I missing? :)

Tellstick firmware is 1.1.1 Stable.

**Comment by hoihrig** [comment]
Yes, Telldus changed the API of their Service slightly. I will release a working version soon

**Comment by hoihrig** [comment]
I made a new release 0.0.5 which does fix this issue for me on 1.1.1 Stable. Please try it out. If you still have an issue please open a new issue.

Thanks for trying it out!

**Comment by senilio** [comment]
Jackpot, everything works as expected now.

Thanks for creating this exporter! I can ditch influxdb now 💃
